Update bonprelevement.class.php

developmentOYR opened the pull request:
Problem


When creating the file sepa ( Direct Debit File) if the amount ends in 0, such as 50.60, the program generates the file with rounding 50.6.


Some banks, for example, La Caixa (CaixaBank) do not accept the direct debit file if the decimal number does not have two digits.


Solution:


What I propose is to create a configuration variable called SEPA_ROUND_TWO_ZERO that allows for a two-digit rounding even if it ends in 0. That is, if the amount is 50.6, the file will be generated with the amount 50.60.

# Instructions
*This is a template to help you make good pull requests. You may use [Github Markdown](https://help.github.com/articles/getting-started-with-writing-and-formatting-on-github/) syntax to format your issue report.*
*Please:*
- *only keep the "FIX", "CLOSE", "NEW", "PERF" or "QUAL" section* (use uppercase to have the PR appears into the ChangeLog, lowercase will not appears)
- *follow the project [contributing guidelines](/.github/CONTRIBUTING.md)*
- *replace the bracket enclosed texts with meaningful information*


# FIX|Fix #[*issue_number Short description*]
[*Long description*]


# CLOSE|Close #[*issue_number Short description*]
[*Long description*]


# NEW|New [*Short description*]
[*Long description*]


# PERF|Perf #[*issue_number Short description*]
[*Long description*]


# QUAL|Qual #[*issue_number Short description*]
[*Long description*]

developmentOYR committed:
Update bonprelevement.class.php

Problem


When creating the file sepa ( Direct Debit File) if the amount ends in 0, such as 50.60, the program generates the file with rounding 50.6.


Some banks, for example, La Caixa (CaixaBank) do not accept the direct debit file if the decimal number does not have two digits.


Solution:


What I propose is to create a configuration variable called SEPA_ROUND_TWO_ZERO that allows for a two-digit rounding even if it ends in 0. That is, if the amount is 50.6, the file will be generated with the amount 50.60.

eldy commented on htdocs/compta/prelevement/class/bonprelevement.class.php:
```diff
@@ -1790,7 +1790,9 @@ public function generate($format = 'ALL', $executiondate = 0, $type = 'direct-de
 				}
-
+				if($conf->global->SEPA_ROUND_TWO_ZERO){
+					$this->total=number_format($this->total,2,".","");
```
We already have a functions to format price using the number if decimal if the country.
Can you try to use price2num with param 'MT' that can be used to force the rounding .

developmentOYR replied:
As you have told me, I have used the price2num function with the 'MT' parameter but I have also had to use the number_format function since this function, when passing a value like 6.50, returns 6.5 and I need it to return 6.50 as i explained to you at the beginning of the request

developmentOYR committed:
Update bonprelevement.class.php

As you have told me, I have used the price2num function with the 'MT' parameter but I have also had to use the number_format function since this function, when passing a value like 6.50, returns 6.5 and I need it to return 6.50 as i explained to you at the beginning of the request

developmentOYR reviewed (commented):
As you have told me, I have used the price2num function with the 'MT' parameter but I have also had to use the number_format function since this function, when passing a value like 6.50, returns 6.5 and I need it to return 6.50 as i explained to you at the beginning of the request

developmentOYR committed:
Merge branch 'develop' into patch-3

developmentOYR committed:
Update bonprelevement.class.php

developmentOYR committed:
Update bonprelevement.class.php

change in format code

developmentOYR committed:
Update bonprelevement.class.php

format in de code

developmentOYR committed:
Update bonprelevement.class.php

format in code

eldy commented:
> As you have told me, I have used the price2num function with the 'MT' parameter but I have also had to use the number_format function since this function, when passing a value like 6.50, returns 6.5 and I need it to return 6.50 as i explained to you at the beginning of the request

Ok, so lets stay on using  "number_format"
However, there is still a lot of errors in code syntax style. I recommend you to install phpcs and phpcbf tool with the git hook pre-commit to have them run and fix your code syntax automatically at each git commit.
See: https://wiki.dolibarr.org/index.php?title=Environment_and_development_tools#Install_Pre-commit_githook

developmentOYR committed:
Update bonprelevement.class.php

format file

developmentOYR committed:
Update bonprelevement.class.php

developmentOYR committed:
Update bonprelevement.class.php

developmentOYR committed:
Update bonprelevement.class.php

developmentOYR committed:
Update bonprelevement.class.php

developmentOYR committed:
Update bonprelevement.class.php

developmentOYR commented:
After several attempts (I couldn't find the key) I managed to get it to pass the tests. There is one that does not pass but fails in a file that I have not modified (htdocs/core/modules/export/export_excel2007.modules.php).
I hope it's enough.

All the best
